x-pack/plugins/cases/public/components/all_cases/table_filter_config/use_filter_config.tsx

```diff
@@ -30,31 +31,68 @@ const mergeSystemAndCustomFieldConfigs = ({
   return newFilterConfig;
 };
 
+const shouldBeActive = ({
+  filter,
+  filterOptions,
+}: {
+  filter: FilterConfigState;
+  filterOptions: FilterOptions;
+}) => {
+  return !filter.isActive && !isEmpty(filterOptions[filter.key as keyof FilterOptions]);
+};
+
+const useActiveByFilterKeyState = ({ filterOptions }: { filterOptions: FilterOptions }) => {
+  const { appId } = useCasesContext();
+  const [activeByFilterKey, setActiveByFilterKey] = useLocalStorage<FilterConfigState[]>(
+    `${appId}.${LOCAL_STORAGE_KEYS.casesTableFiltersConfig}`,
+    []
+  );
+
+  /**
+   * Activates filters that aren't active but have a value in the filterOptions
+   */
+  const newActiveByFilterKey = [...(activeByFilterKey || [])];
+  newActiveByFilterKey.forEach((filter) => {
+    if (shouldBeActive({ filter, filterOptions })) {
+      const currentIndex = newActiveByFilterKey.findIndex((_filter) => filter.key === _filter.key);
+      newActiveByFilterKey.splice(currentIndex, 1);
+      newActiveByFilterKey.push({ key: filter.key, isActive: true });
+    }
+  });
+
+  if (!isEqual(newActiveByFilterKey, activeByFilterKey)) {
+    setActiveByFilterKey(newActiveByFilterKey);
+  }
+
+  return [newActiveByFilterKey, setActiveByFilterKey] as [
+    FilterConfigState[],
```

Comment:
Why do we have to cast here?

Reply:
because otherwise the type will be set to Array<FilterConfigState[] |  (value: SetStateAction<FilterConfigState[]>) => void> which isn't right and will throw ts errors when destructuring